docs/modules/ROOT/pages/explanations/dns_scheme.adoc

```diff
@@ -0,0 +1,121 @@
+= DNS Scheme
+
+An OpenShift 4 cluster requires using a specific DNS scheme which is described on this page. Most of the values are automatically generated and can't be changed.
+
+== User selectable values
+
+`<base domain>`::
+
+  The base domain for the cluster (input by the customer).
+
+== Automatically generated values
+
+`<cluster id>`::
+
+  The Project Syn Lieutenant cluster id of the cluster (auto generated by the Lieutenant API).
+
+`<role>`::
+
+  Machine role. One of _master_, _infra_, _worker_ and _lb_. Other roles possible when required by the customer (special node groups for builds, GPU support etc.).
+
+`<#>`::
+
+  A automatically randomly defined character sequence with four digits.
+
+`<index>`::
+
+  A zero indexed incrementing number.
+
+`<cluster domain>`::
+
+  `<cluster id>.<base domain>`
+
+`<hostname>`::
+
+  The fully qualified domain name of machines belonging to the cluster.
+
+  `<role><#>.<cluster domain>`
+
+`<external api>`::
+
+  OpenShift API URL to be used by clients external to the cluster.
+
+  `api.<cluster domain>`
+
+`<internal api>`::
+
+  OpenShift API URL to be used by clients internal to the cluster.
+
+  `api-int.<cluster domain>`
+
+`<default app domain>`::
+
+  `*.apps.<cluster domain>`
+
+`<etcd>`::
+
+  `etcd-<index>.<cluster domain>`
+
+`<etcd srv>`::
+
+  DNS Resource Record of type `SRV` pointing to the `<etcd>` instances.
+
+  `_etcd-server-ssl._tcp.<cluster domain>`
+
+`<acme challenge>`::
+
+  DNS Resource Record of type `TXT` used for validating certificates using the DNS01 method.
+  *It must be possible to create these records programmatically with one of the APIs supported by https://cert-manager.io/docs/configuration/acme/dns01/#supported-dns01-providers[cert-manager]*.
+
+  `_acme-challenge.<default app domain | external api>`
+
+[IMPORTANT]
+====
+The following DNS records must resolve publicly:
+
+ * `<acme challenge>`
+ * `<cluster domain>`
+ * `<default app domain>`
+ * `<external api>`
+
+ Everything else must resolve at least within the clusters network.
+====
+
+== Example
+
+A full list of DNS records for one cluster using the above definition:
+
+----
+; Hosts
+infra-454b.c-summer-shadow-2592.example.com IN A x.x.x.x
```

Comment:
Records for the nodes are not technically required, except for the etcd (master) nodes. Should we still create/require them?
It's not clear to me what the benefit/drawback of it would be...

Reply:
For me we don't really need DNS names for nodes, I'm not sure where this is coming from.

Reply:
I guess this is an old habit. I would be fine not having those records. We still have not defined if and how we access machines belonging to the cluster. Depending on that, the records will be a convenience or even a requirement. As for the LBs and if I am not misstaken, we need the records in order for SSHop to work (not now but I am thinking ahead).

Reply:
We do not need the DNS records for now. However, I see the need for having their naming defined. I removed them from the DNS example and listed them separately. In addition, I noted that `<hostname>` is only for naming and will not be used in DNS.